### PR TITLE
fix available urls in api

### LINKED ISSUE #638: API: available_language_urls wrong
Example:
en | "http://integreat/wordpress/augsburg/en/lebensorganisation/freizet/buecherei/"
ar | "http://integreat/wordpress/augsburg/ar/lebensorganisation/freizet/buecherei/"
fa | "http://integreat/wordpress/augsburg/fa/lebensorganisation/freizet/buecherei/"

### PR DESCRIPTION
**Pull request by timoludwig**
use the working filter ap_post_to_foreign_language_id to check whether the available_language_url makes sense. (fixes #638)

### CODE REVIEW

File: wp-content/plugins/ig-wp-api-extensions/endpoints/v0/RestApi_ModifiedContent.php [hunk 1]
```diff
@@ -230,6 +234,7 @@ protected function build_query_order_clauses() {
 		return ["menu_order ASC", "post_title ASC"];
 	}
 
+	/* Probably unused function. available_languages uses the function in line 6 */
```
Comment (svenseeberg) [inline wp-content/plugins/ig-wp-api-extensions/endpoints/v0/RestApi_ModifiedContent.php]: I agree. Let's remove the function. Unless @maxammann you had some other intentions here?

File: wp-content/plugins/ig-wp-api-extensions/endpoints/v0/RestApi_ModifiedContent.php [hunk 2]
```diff
@@ -241,6 +246,7 @@ protected function map_post_to_foreign_language_id($post, $language_code) {
 		return $id;
 	}
 
+	/* Probably unused function. available_language_urls uses the function in line 14 */
```
Comment (svenseeberg) [inline wp-content/plugins/ig-wp-api-extensions/endpoints/v0/RestApi_ModifiedContent.php]: I agree. Let's remove the function. Unless @maxammann you had some other intentions here?

File: wp-content/plugins/ig-wp-api-extensions/endpoints/v0/RestApi_ModifiedContent.php (hunk 1 above)
Comment (maxammann) [reply wp-content/plugins/ig-wp-api-extensions/endpoints/v0/RestApi_ModifiedContent.php]: You can remove them. They don't seem to be used. Those were my functions and you changed them, right?

Comment (svenseeberg) [reply wp-content/plugins/ig-wp-api-extensions/endpoints/v0/RestApi_ModifiedContent.php]: that is correct.

**Comment by timoludwig** [comment]
@svenseeberg thanks for your feedback. The only error I could reproduce was delivering URLs for non-existing translations. Could you give an example where there's another issue?

**Comment by svenseeberg** [comment]
I think that you're correct, it was mostly a gut feeling. But I'll check later today. Thanks for your work!

**Comment by svenseeberg** [comment]
Okay, there is still a bug. I created a new german page with the permalink http://integreat/wordpress/augsburg/de/testseite/ and created an english translation with the permalink http://integreat/wordpress/augsburg/en/test-page/

Now the content of http://integreat/wordpress/augsburg/de/wp-json/extensions/v0/modified_content/pages?since=2015-01-25T09%3A27%3A49%2B0000 is:
`[{"id":8146,"permalink":{"url":"http:\/\/integreat\/wordpress\/augsburg\/de\/testseite\/","url_site":"http:\/\/integreat\/wordpress\/augsburg","url_page":"testseite","url_page_id":"http:\/\/integreat\/wordpress\/augsburg\/?page_id=8146","url_date_1_name":"http:\/\/integreat\/wordpress\/augsburg\/2018\/01\/13\/testseite","url_date_2_name":"http:\/\/integreat\/wordpress\/augsburg\/2018\/01\/testseite"},"title":"Testseite","type":"page","status":"publish","modified_gmt":"2018-01-13 20:41:53","excerpt":"Testinhalt","content":"<p>Testinhalt<\/p>\n","parent":0,"order":0,"available_language_urls":{"en":"http:\/\/integreat\/wordpress\/augsburg\/en\/testseite\/"},"available_languages":{"en":8148},"thumbnail":null,"automatic_translation":""}]`

And the content of http://integreat/wordpress/augsburg/en/wp-json/extensions/v0/modified_content/pages?since=2015-01-25T09%3A27%3A49%2B0000 :
`[{"id":8148,"permalink":{"url":"http:\/\/integreat\/wordpress\/augsburg\/en\/test-page\/","url_site":"http:\/\/integreat\/wordpress\/augsburg","url_page":"test-page","url_page_id":"http:\/\/integreat\/wordpress\/augsburg\/?page_id=8148","url_date_1_name":"http:\/\/integreat\/wordpress\/augsburg\/2018\/01\/13\/test-page","url_date_2_name":"http:\/\/integreat\/wordpress\/augsburg\/2018\/01\/test-page"},"title":"Test page","type":"page","status":"publish","modified_gmt":"2018-01-13 20:42:21","excerpt":"Test content.","content":"<p>Test content.<\/p>\n","parent":0,"order":0,"available_language_urls":{"de":"http:\/\/integreat\/wordpress\/augsburg\/de\/test-page\/"},"available_languages":{"de":8146},"thumbnail":null,"automatic_translation":""}]`

**Pull request merged**